File: Best Sightseeing Pair.cpp
class Solution {
public:
    int maxScoreSightseeingPair(vector<int>& values) {
        int maxx0 = 0, maxx1 = 0;
        for (int val: values) {
            maxx1 = max(maxx1, val+maxx0-1);
            maxx0 = max(maxx0-1, val);
        }
        return maxx1;
    }
};

File: Best Sightseeing Pair.py
class Solution:
    def maxScoreSightseeingPair(self, values: List[int]) -> int:
        maxx0, maxx1 = 0, 0
        for val in values:
            maxx0, maxx1 = max(maxx0-1, val), max(maxx1, val+maxx0-1)
        return maxx1
